### PR TITLE
Add the ability to attach validation to a table

### PR DESCRIPTION
**Pull request by rsheeter**
And use it on fvar and prove it works.

### CODE REVIEW

File: write-fonts/src/tables/fvar.rs [hunk 1]
```diff
@@ -8,6 +8,19 @@ pub use instance_record::InstanceRecord;
 include!("../../generated/generated_fvar.rs");
 
 impl Fvar {
+    /// We need everyone to have, or not have, post_script name so we can have a single record size.
+    pub fn check_instance_size(&self, ctx: &mut ValidationCtx) {
```
Comment (dfrg) [inline write-fonts/src/tables/fvar.rs:12]: @cmyr do these need to be public?

File: write-fonts/src/tables/fvar.rs [hunk 2]
```diff
@@ -8,6 +8,19 @@ pub use instance_record::InstanceRecord;
...
+        let sum: i32 = self
+            .axis_instance_arrays
+            .instances
+            .iter()
+            .map(|ir| ir.post_script_name_id.map(|_| 1).unwrap_or(-1))
```
Comment (dfrg) [inline write-fonts/src/tables/fvar.rs:17]: Cleverness noted :)

While we’re here, we should probably also check that `instance.coordinates.len() == axis_count` for all instances.

**Review by dfrg — APPROVED** [review]
Looks good with a suggestion for an additional check.

File: write-fonts/src/tables/fvar.rs (hunk 1 above)
Comment (rsheeter) [reply write-fonts/src/tables/fvar.rs:12]: That's a good point, let me see if anything kerplodes if not.

Comment (rsheeter) [reply write-fonts/src/tables/fvar.rs:12]: It appears they do not, unless there's some reason someone would call them from The Outside. For now I will update to not be.

**Pull request merged**